third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -79,13 +81,26 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
   std::vector<ComputationClient::DataPtr> datas;
   datas.reserve(tensors.size());
   for (auto& tensor : tensors) {
-    xla::Literal literal(tensor.shape);
-    tensor.populate_fn(tensor, literal.untyped_data(), literal.size_bytes());
-
     PjRtDevice* pjrt_device = StringToPjRtDevice(tensor.device);
-    std::shared_ptr<xla::PjRtBuffer> buffer =
-        client_->BufferFromHostLiteral(literal, pjrt_device).ValueOrDie();
-    buffer->GetReadyFuture().Await();
```

Comment:
Why is this step no longer needed under `BufferFromHostBuffer`?

Reply:
This makes sense now.

Reply:
`Await` was used before to ensure that `literal` (which was on the stack) was kept alive for the duration of the transfer. Using this new implementation, a callback takes ownership of `literal`, and after that callback fires (after the transfer is complete), the `literal` gets freed. So, basically, that means we can perform the transfer asynchronously without the risk of `literal` getting cleaned up in the middle of the transfer.